### DISCUSSION TITLE
Running NATS server on isolated CPU cores

### DISCUSSION THREAD

**Discussion by EgusCpp** [nats-io/nats-server#4863] [Q&A]
Greetings!  
  
On the Linux, we have an option to isolate a set of CPU cores from the system, to let performance-critical app to be running on these cores exclusively. Is it possible to force NATS server to run only on selected cores?   
  
Moreover, is there any way to divide streams and KV among cores? Lets's say we have streams A,B,C which could be quite intensive while not latency-critical, and a set of KV buckets D,E,F - which are latency-critical. So idea is to ensure A,B,C would use let's say cores 1,2,3 and D,E,F will use cores 4,5,6, so they won't impact each other (at least in terms of CPU time and context switching).   
   
Is it all ever applicable to NATS server architecture?

**Comment by derekcollison** [comment]
The nats-server is a native binary, so anything applicable to normal binaries or containers can be applied to the nats-server.

**Reply by EgusCpp** [reply]
In other words, you mean we just need to launch NATS server using "taskset" command, to let it run on specific core, and there is no option to assign cores internally? If so, then it seems if I need to run some tasks on one CPU core and others on another CPU core, I need to configure a set of running NATS binaries?
